adapters/myx/src/sdk/positionSnapshots.ts

```diff
@@ -0,0 +1,146 @@
+    timestamp: number
+}
+
+interface SubgraphResponse {
+    data: {
+        userLPStats: LiquidityPositionSnapshot[]
+    }
+}
+
+export const getPositionsForAddressByPoolAtBlock = async (
+    snapshotBlockNumber: number
+): Promise<OutputDataSchemaRow[]> => {
+    const userPositionSnapshotsAtBlockData: OutputDataSchemaRow[] = []
+    let snapshotsArrays: LiquidityPositionSnapshot[] = []
+    const snapshotsMap = new Map<string, Map<string, LiquidityPositionSnapshot>>() // user => pool => snapshot
+    let skip = 0
+    const b_end = snapshotBlockNumber
+    let b_start = 0
+    // eslint-disable-next-line no-constant-condition
+    while (true) {
+        let query = `
+             query filterSnapshots {
+                  userLPStats (
+                  skip: ${skip},
+                    first: 1000,
+                    orderBy: block, 
+                    orderDirection: asc,
+                    where: {
+                      block_gt: ${b_start},
+                      block_lte: ${b_end},
+                    }
+                  ) {
+                        lPToken {
+                          address
+                          symbol
+                        }
+                        token0 {
+                          address
+                          symbol
+                        }
+                        token1 {
+                          address
+                          symbol
+                        }
+                        pairIndex
+                        recipient
+                        lpAmount
+                        token0Amount
+                        token1Amount
+                        timestamp
+                      }
+                }
+            `
+        const res = await fetch(SUBGRAPH_URLS[CHAINS.LINEA], {
+            method: "POST",
+            body: JSON.stringify({query}),
+            headers: {"Content-Type": "application/json"},
+        }).then(res => res.json()) as SubgraphResponse
+        snapshotsArrays = snapshotsArrays.concat(res.data.userLPStats)
+        if (res.data.userLPStats.length !== 1000) {
+            break
+        }
+        skip += 1000
+        if (skip > 5000) {
+            skip = 0
+            b_start = snapshotsArrays[snapshotsArrays.length - 1].block + 1
+        }
+        writeProgress(b_end, b_start, b_end)
+    }
+    for (const snapshot of snapshotsArrays) {
+        let userPositionSnapshotMap = snapshotsMap.get(snapshot.recipient)
+        if (!userPositionSnapshotMap) {
+            userPositionSnapshotMap = new Map<string, LiquidityPositionSnapshot>()
+        }
+        userPositionSnapshotMap.set(snapshot.lPToken.address, snapshot)
+        snapshotsMap.set(snapshot.recipient, userPositionSnapshotMap)
+    }
+    snapshotsMap.forEach((userPositionSnapshotMap => {
+        userPositionSnapshotMap.forEach((positionSnapshot) => {
+            userPositionSnapshotsAtBlockData.push({
+                user_address: positionSnapshot.recipient,
```

Comment:
Will this return multiple block numbers for 1 block input?

Same question for timestamp as well
@marvin6en

Reply:
at line 55-58，the graphql Already filtered